integration-tests/tests/src/tests/queries.ts

```diff
@@ -15,129 +15,1031 @@
+          [3, "stringCol BEGINSWITH 'a'"],
+          [1, "stringCol beginswith 'ab'"],
+          [0, "stringCol BEGINSWITH 'abcd'"],
+          [2, "stringCol BEGINSWITH 'A'"],
+          [2, "stringCol ENDSWITH 'c'"],
+          [1, "stringCol endswith 'bc'"],
+          [9, "stringCol ENDSWITH ''"],
+          [1, "stringCol CONTAINS 'b'"],
+          [2, "stringCol contains 'c'"],
+          [9, "stringCol CONTAINS ''"],
+          [2, "stringCol == $0", "a"],
+          [2, "stringCol ENDSWITH $0", "c"],
+          [2, "stringCol BEGINSWITH 'a' LIMIT(2)"],
+          [3, "stringCol ==[c] 'a'"],
+          [5, "stringCol BEGINSWITH[c] 'A'"],
+          [4, "stringCol ENDSWITH[c] 'c'"],
+          [2, "stringCol CONTAINS[c] 'B'"],
+
+          [1, "stringCol == $0", null],
+          [1, "$0 == stringCol", null],
+          [9, "stringCol != $0", null],
+
+          [1, "stringCol == null"],
+          [1, "null == stringCol"],
+          [9, "stringCol != null"],
+        ]);
+      });
+
+      it("throws with invalid queries", function (this: RealmContext) {
+        assertQueryException(this.realm, NullableTypesObject, [
+          ["Unsupported comparison between type", "stringCol == true"],
+          ["Unsupported comparison between type", "stringCol == 123"],
+          ["Unsupported comparison operator", "stringCol CONTAINS $0", 1],
+        ]);
+      });
+    });
+
+    describe("querying with data", () => {
+      let objects: NullableTypesObject[];
+      beforeEach(function (this: RealmContext) {
+        objects = this.realm.write(() => {
+          return [
+            this.realm.create<INullableTypesObject>("NullableTypesObject", {
+              dataCol: new Uint8Array([1, 100, 233, 255, 0]),
+            }),
+            this.realm.create<INullableTypesObject>("NullableTypesObject", { dataCol: new Uint8Array([1, 100]) }),
+            this.realm.create<INullableTypesObject>("NullableTypesObject", { dataCol: new Uint8Array([100]) }),
+            this.realm.create<INullableTypesObject>("NullableTypesObject", { dataCol: new Uint8Array([2]) }),
+            this.realm.create<INullableTypesObject>("NullableTypesObject", { dataCol: new Uint8Array([255, 0]) }),
+            this.realm.create<INullableTypesObject>("NullableTypesObject", { dataCol: undefined }),
+          ];
+        });
+      });
+
+      it("returns correct length", function (this: RealmContext) {
+        assertQueryLength(this.realm, NullableTypesObject, [
+          [1, "dataCol == $0", objects[1].dataCol],
+          [1, "$0 == dataCol", objects[2].dataCol],
+          [5, "dataCol != $0", objects[0].dataCol],
+          [1, "dataCol BEGINSWITH $0", objects[0].dataCol],
+          [2, "dataCol BEGINSWITH $0", objects[1].dataCol],
+          [2, "dataCol ENDSWITH $0", objects[4].dataCol],
+          [3, "dataCol CONTAINS $0", objects[2].dataCol],
+
+          [1, "dataCol == null"],
+          [5, "dataCol != $0", null],
+        ]);
+      });
```

Comment:
Maybe make a todo for the types that are missing "invalid queries".  Currently they are missing for objectId and data.  Perhaps there are others.